java/openmldb-taskmanager/src/main/java/com/_4paradigm/openmldb/taskmanager/client/TaskManagerClient.java

```diff
@@ -35,17 +44,106 @@
+        final NodeCache nodeCache = new NodeCache(curator, path, false);
+        class NodeListener implements NodeCacheListener {
+            RpcClient rpcClients;
+            RpcClientOptions clientOption;
+
+            public NodeListener(RpcClient rpcClients, RpcClientOptions clientOption) {
+                this.rpcClients = rpcClients;
+                this.clientOption = clientOption;
+            }
+
+            @Override
+            public void nodeChanged() throws Exception {
+                String endpoint = new String(nodeCache.getCurrentData().getData());
+                if (endpoint != null) {
+                    rpcClient.stop();
+                    logger.info("The content of the node was changed, try to reconnect");
+                    rpcClient.stop();
+
+                    String serviceUrl = "list://" + endpoint;
+                    rpcClient = new RpcClient(serviceUrl, clientOption, new ArrayList<Interceptor>());
+                    taskManagerInterface = BrpcProxy.getProxy(rpcClient, TaskManagerInterface.class);
+                    RpcContext.getContext().setLogId(1234L);
+                } else {
+                    logger.info("The content of the node was deleted, please try to reconnect");
+                    close();
+                }
+            }
+        }
+        nodeCache.getListenable().addListener(new NodeListener(rpcClients, clientOption));
+        nodeCache.start(true);
+    }
 
     /**
      * Constructor of TaskManager client.
      *
      * @param endpoint the endpoint of TaskManager server, for example "127.0.0.1:9902".
      */
-    public TaskManagerClient(String endpoint) {
+    public void rpcConnect(String endpoint) {
 
+        clientOption = new RpcClientOptions();
+        clientOption.setProtocolType(Options.ProtocolType.PROTOCOL_BAIDU_STD_VALUE);
+        clientOption.setWriteTimeoutMillis(1000);
+        clientOption.setReadTimeoutMillis(50000);
+        clientOption.setMaxTotalConnections(1000);
+        clientOption.setMinIdleConnections(10);
+        clientOption.setLoadBalanceType(LoadBalanceStrategy.LOAD_BALANCE_FAIR);
+        clientOption.setCompressType(Options.CompressType.COMPRESS_TYPE_NONE);
+
+        String serviceUrl = "list://" + endpoint;
+        List<Interceptor> interceptors = new ArrayList<Interceptor>();
+        if (rpcClient != null) {
+            this.stop();
+        }
+        rpcClient = new RpcClient(serviceUrl, clientOption, interceptors);
+        taskManagerInterface = BrpcProxy.getProxy(rpcClient, TaskManagerInterface.class);
+        RpcContext.getContext().setLogId(1234L);
+    }
+
+    /**
+     * Constructor of TaskManager client.
+     *
+     * @param endpoint the endpoint of TaskManager server, for example "127.0.0.1:9902".
+     */
+    public TaskManagerClient(String endpoint) {
 
-        RpcClientOptions clientOption = new RpcClientOptions();
+        clientOption = new RpcClientOptions();
```

Comment:
why not reuse `rpcConnect()` in `public TaskManagerClient(String endpoint)`?

Reply:
Ignore the modification just now, it should be changed to the rpcConnect method (forget the rpcConnect method has been changed last week)